### PR TITLE
Fix gen jet variables, add new jet variables

### PR DESCRIPTION
**Pull request by riga**
This PR fixes the lookup of the gen jet index per `pat::Jet`, and adds a jet variables which seem helpful.

Fixed indices:
- Before the index was obtained by comparing pointers, but since the gen jet lookup creates a copy, the pointers will never match any object in the actual gen jet collection.
- The fix uses a feature of the forward reference, which already stores the index.

New variables:
- `bParticleNetAK4JetTags_probbb`
- `bParticleNetAK4JetTags_probpu`
- `bParticleNetAK4JetTags_probcc`
- `bParticleNetAK4JetTags_probundef`
- `bParticleNetAK4JetTags_probc`
- `bParticleNetAK4JetTags_probb`
- `bParticleNetAK4JetTags_probuds`
- `bParticleNetAK4JetTags_probg`
- `ak8jets_massIndependentDeepDoubleBvLJetTags_probHbb`
- `ak8jets_deepDoubleBvLJetTags_probHbb`
- `ak8jets_deepBoostedJetTags_probHbb`
- `ak8jets_particleNetJetTags_probHbb`
- `ak8jets_particleNetDiscriminatorsJetTags_HbbvsQCD`

Edit: Sorry, I saw only now that my editor converted places that had mixed spaces/tabs into just spaces. I can revert that if needed.

### CODE REVIEW

**Comment by dzuolo** [comment]
@riga : I assume this PR has been tested with the whole 2017 production right? ;) If there is nothing else you would like to add i propose we merge it. Do you agree @portalesHEP ?

**Comment by portalesHEP** [comment]
I was thinking of waiting for some preliminary results to be sure we'll use the new variables, but it's fine for me either way

**Comment by riga** [comment]
> @riga : I assume this PR has been tested with the whole 2017 production right? ;) If there is nothing else you would like to add i propose we merge it. Do you agree @portalesHEP ?

Yes, we ran the whole 2017 bigNtuple production with it, but we haven't looked into their distributions yet. We we're planning to do this with the small ntuples asap, and we bring them into one of the next meetings.

**Comment by portalesHEP** [comment]
Following up on this morning's discussions, since we'll very likely want to study ParticleNet for future analyses as well I guess it does not hurt to have the variables included already even if we end up not using it in the current analysis (although I guess the variables and framework may change in the meanwhile). So if nothing else should be added, I'll merge

**Comment by dzuolo** [comment]
I agree with you Louis, if these variables are ready to be used I propose to add them!

File: NtupleProducer/plugins/HTauTauNtuplizer.cc [hunk 1]
```diff
@@ -2338,6 +2378,16 @@ int HTauTauNtuplizer::FillJet(const edm::View<pat::Jet> *jets, const edm::Event&
...
+    // ParticleNet
+    _bdiscr15.push_back(ijet->bDiscriminator("pfParticleNetAK4JetTags:probbb"));
+    _bdiscr16.push_back(ijet->bDiscriminator("pfParticleNetAK4JetTags:probpu"));
+    _bdiscr17.push_back(ijet->bDiscriminator("pfParticleNetAK4JetTags:probcc"));
+    _bdiscr18.push_back(ijet->bDiscriminator("pfParticleNetAK4JetTags:probundef"));
+    _bdiscr19.push_back(ijet->bDiscriminator("pfParticleNetAK4JetTags:probc"));
+    _bdiscr20.push_back(ijet->bDiscriminator("pfParticleNetAK4JetTags:probb"));
+    _bdiscr21.push_back(ijet->bDiscriminator("pfParticleNetAK4JetTags:probuds"));
+    _bdiscr22.push_back(ijet->bDiscriminator("pfParticleNetAK4JetTags:probg"));
+
```
Comment (francescobrivio) [inline NtupleProducer/plugins/HTauTauNtuplizer.cc:2390]: Just double checking: why are these called `AK4` when they are used for AK8 jets?

Comment (dzuolo) [reply NtupleProducer/plugins/HTauTauNtuplizer.cc:2390]: I learned in the talk by Raffa at our MIB meeting that ParticleNet can tag also resolved jets ;)

Comment (francescobrivio) [reply NtupleProducer/plugins/HTauTauNtuplizer.cc:2390]: Ok cool" Thanks Davide! My question was more about the "correctness" of the names of the taggers?

Comment (dzuolo) [reply NtupleProducer/plugins/HTauTauNtuplizer.cc:2390]: I assume these are the taggers for AK4 jets, the one for AK8 have AK8 in the name...

Comment (portalesHEP) [reply NtupleProducer/plugins/HTauTauNtuplizer.cc:2390]: I'll let @riga confirm, but these names look correct (and are defined [here](https://github.com/cms-sw/cmssw/blob/CMSSW_10_6_X/RecoBTag/ONNXRuntime/python/pfParticleNetAK4_cff.py))

**Comment by dzuolo** [comment]
One question that came to my mind: for the AK8 jets (maybe there is something similar also for AK4) is the regressed mass also available? I know it has been used in a couple of boosted HH analysis.

**Comment by portalesHEP** [comment]
> One question that came to my mind: for the AK8 jets (maybe there is something similar also for AK4) is the regressed mass also available? I know it has been used in a couple of boosted HH analysis.

~~To my understanding, the regressed mass was obtained by "hacking" ParticleNet (removing the output layer iirc, probably in addition to some more changes)~~
It'd seem that the regression was included (at least) in NanoAODs (cf. https://github.com/cms-sw/cmssw/pull/33588). If it's also available in MiniAODs it could be interesting to look into as well indeed

**Comment by dzuolo** [comment]
I have the same recollection, I was just proposing to check if something similar was added to miniAOD

**Comment by riga** [comment]
This has been a while ... 😇

I just checked again and it's does not seem to be added on mini-level.

**Comment by portalesHEP** [comment]
Indeed this was left hanging. Can it (/should it)  be merged now or are there further tests being considered?

**Comment by riga** [comment]
I re-checked the event content shipped with the miniAOD files and I think the code in PR should cover everything that we might need, so from our side the PR could be merged 👍

**Pull request merged**